bigbluebutton-html5/imports/ui/components/presentation/component.jsx

```diff
@@ -1,78 +1,246 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import CSSTransitionGroup from 'react-transition-group/CSSTransitionGroup';
+import WhiteboardOverlayContainer from '/imports/ui/components/whiteboard/whiteboard-overlay/container';
+import WhiteboardToolbarContainer from '/imports/ui/components/whiteboard/whiteboard-toolbar/container';
 import PollingContainer from '/imports/ui/components/polling/container';
+import CursorWrapperContainer from './cursor/cursor-wrapper-container/container';
 import AnnotationGroupContainer from '../whiteboard/annotation-group/container';
-import Cursor from './cursor/component';
 import PresentationToolbarContainer from './presentation-toolbar/container';
+import PresentationOverlayContainer from './presentation-overlay/container';
 import Slide from './slide/component';
 import styles from './styles.scss';
 
+
 export default class PresentationArea extends React.Component {
-  constructor(props) {
-    super(props);
+  constructor() {
+    super();
+
+    this.state = {
+      paperWidth: 0,
+      paperHeight: 0,
+      showSlide: false,
+    };
+
+    this.getSvgRef = this.getSvgRef.bind(this);
   }
 
+  componentDidMount() {
+    // adding an event listener to scale the whiteboard on 'resize' events sent by chat/userlist etc
+    window.addEventListener('resize', () => {
+      setTimeout(this.handleResize.bind(this), 0);
+    });
+
+    this.getInitialPaperSizes();
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', () => {
+      setTimeout(this.handleResize.bind(this), 0);
+    });
+  }
+
+  // returns a ref to the svg element, which is required by a WhiteboardOverlay
+  // to transform screen coordinates to svg coordinate system
+  getSvgRef() {
+    return this.svggroup;
+  }
+
+  getPaperSizes() {
+    const { presentationPaper, whiteboardSizeAvailable } = this;
+    const paperSizes = {};
+
+    if (presentationPaper) {
+      // if a user is a presenter - this means there is a whiteboardToolBar on the right
+      // and we have to get the width/height of the whiteboardSizeAvailable
+      // (inner hidden div with absolute position)
+      let clientHeight;
+      let clientWidth;
+      if (this.props.userIsPresenter) {
+        clientHeight = whiteboardSizeAvailable.clientHeight;
+        clientWidth = whiteboardSizeAvailable.clientWidth;
+      // user is not a presenter - we can get the sizes of the presentationPaper
+      // direct parent of the svg wrapper
+      } else {
```

Comment:
Unnecessary only if we initialize with 
```
clientHeight = presentationPaper.clientHeight;
clientWidth = presentationPaper.clientWidth;
```

instead of 
```
 let clientHeight;
 let clientWidth;
```

Is this what you meant?

Reply:
@antobinary yes